src/__tests__/utils/arithmetic-helper.test.ts

```diff
@@ -0,0 +1,490 @@
+      const outputParameter = '= result + 10';
+      const output = { '= result + 10': 15 };
+  
+      const result = evaluateArithmeticOutput(outputParameter, output);
+  
+      expect(result).toEqual({ result: 25 });
+    });
+  })
+
+  describe('evaluateInput(): ', () => {
+    it('should return the same input when no expressions are present.', () => {
+      const input = { param1: 10, param2: 20 };
+      
+      const result = evaluateInput(input);
+  
+      expect(result).toEqual({ param1: 10, param2: 20 });
+    });
+
+    it('should evaluate simple arithmetic expression for a parameter.', () => {
+      const input = { param1: '=2*5' };
+      
+      const result = evaluateInput(input);
+  
+      expect(result).toEqual({ param1: 10 });
+    });
+  
+    it('should evaluate multiple parameters with arithmetic expressions', () => {
+      const input = { param1: '=2*5', param2: '=3+7' };
+      
+      const result = evaluateInput(input);
+  
+      expect(result).toEqual({ param1: 10, param2: 10 });
+    });
+  
+    it('should evaluate parameters with dependencies on other parameters', () => {
+      const input = { param1: 10, param2: '=param1*2' };
+      
+      const result = evaluateInput(input);
+  
+      expect(result).toEqual({ param1: 10, param2: 20 });
+    });
+  
+    it('should evaluate nested dependencies', () => {
+      const input = { param1: 10, param2: '=param1*2', param3: '=param2+5' };
+      
+      const result = evaluateInput(input);
+  
+      expect(result).toEqual({ param1: 10, param2: 20, param3: 25 });
+    });
+  
+    it('should return original value if expression is invalid', () => {
+      const input = { param1: '=invalid*expression' };
+      
+      expect(() => {
+        evaluateInput(input);
+      }).toThrow(InputValidationError)
+    });
+  
+    it('should handle a mix of valid expressions, dependencies, and plain values', () => {
+      const input = { param1: 10, param2: '=param1+5', param3: 30, param4: '=param3*2' };
+      
+      const result = evaluateInput(input);
+  
+      expect(result).toEqual({ param1: 10, param2: 15, param3: 30, param4: 60 });
+    });
+
+    it('should throw an error for unsupported arithmetic operator in expression.', () => {
+      const input = { param1: '=2%5' };
+    
+      expect(() => evaluateInput(input)).toThrow(
+        new WrongArithmeticExpressionError(
+          'The operator in `=2%5` should be one of these arithmetic operators: *, +, - or /.'
+        )
+      );
+    });
+
+    it('should throw an error if a parameter value is not a number.', () => {
+      const input = { param1: '=param2*2', param2: 'mock-param' }; 
+    
+      expect(() => evaluateInput(input)).toThrow(
+        new InputValidationError(
+          'The value of the `param2` parameter in the input array is not a number.'
+        )
+      );
+    });
+
+    // it('should throw an error when input contains division by zero.', () => {
```

Comment:
you can uncomment it, now it should work. Just change the error message to this:
'The input expression contains a division by zero: `10/0`.'